tests/unit/deferred.py

```diff
@@ -96,17 +105,9 @@ def modify_port(self, port, channel, network_id):
 
             # setup a different connection to database so that this method does
             # not see uncommited changes by `apply_networking`
-
-            local_app = Flask(__name__.split('.')[0])
-            uri = config.cfg.get('database', 'uri')
-            local_app.config.update(SQLALCHEMY_TRACK_MODIFICATIONS=False)
-            local_app.config.update(SQLALCHEMY_DATABASE_URI=uri)
-            local_db = SQLAlchemy(local_app)
-
-            current_count = local_db.session \
-                .query(model.NetworkingAction).count()
-
-            local_db.session.commit()
+            with app.app_context():
```

Comment:
Honestly, now that I think about it I like @naved001's original solution better than either. The problem with `app_context` is that here we have *nested* app contexts, and I can't find any documentation talking about that situation at all; I wouldn't want to just assume it's valid, although it may be. The issue with `scoped_session` is that I'm not sure where `Flask-SQLAlchemy` stores our session factory, and more generally, the details of that sort of integration are poorly documented. The original code is clearly correct without having to delve into arcane details of how the frameworks operate; that's worth a few extra lines of code (wish I'd spoken up a bit earlier).